introduce package sorting in the timetable

SharglutDev opened the pull request:
close #14305 

> [!TIP]
> Second commit is just moving code, no functional change

SharglutDev committed:
front: add timetable mode switch

- Reorder timetable toolbar icons
- Add timetable mode switch

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

SharglutDev committed:
front: extract timetable train list in its own component

This will be usefull to display a list of trains in the package mode
tabs.

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

Caracol3 commented on front/src/styles/scss/applications/operationalStudies/_scenario.scss:
```diff
@@ -575,45 +575,75 @@
+              align-items: center;
+              background-color: var(--black5);
+              border-radius: 4px;
+              margin-bottom: -5px; // to fix icon weird height
```
To match the mockup, maybe we can change the `padding-top` from `5px` to `6px` until we fix the icon height issue.

SharglutDev replied:
Hmm not sure when you want the `padding-top` ? I can't find a way to fix the position by using this.

Caracol3 replied:
Did i say padding-top ?? 🤦‍♂️
Sorry, i meant the `margin-bottom` from `-5px` to `-6px` maybe.

SharglutDev replied:
Haha ok, done :)

Caracol3 reviewed (commented):
LGTM (tested) for now.
Just left one comment.

SharglutDev committed:
front: refacto menu trigger button styles

The styles were linked to the boardwrapper ones but until now the design system uses the same ones for every instance of the menu trigger button.
Move the styles in a common style file and adapt the component so it can still receive custom styles.

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

SharglutDev committed:
front: display sandbox in timetable package mode

Signed-off-by: SharglutDev <p.filimon75@gmail.com>

kmer2016 reviewed (approved):
LGTM! Good job
